scripts/js/commands/checkNewPills.ts

```diff
@@ -0,0 +1,99 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2025.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import fs from "fs/promises";
+import path from "path";
+
+import { globby } from "globby";
+import yargs from "yargs/yargs";
+import { hideBin } from "yargs/helpers";
+import { flatten, flattenDeep } from "lodash-es";
+
+import { TocEntry } from "../lib/api/generateToc.js";
+
+const TODAY = new Date();
+
+interface NewPillEntry {
+  url: string;
+  date?: string | "null";
```

Comment:
```suggestion
  date: string | null;
```

1. `null` is a special value, whereas you're using the string "null" right now
2. `date?` is shorthand for `date: string | "null" | undefined`. JavaScript is weird that it has both `undefined` and `null`. We only want to use `null` to model that the date is missing; it's confusing to use both `undefined` and `null`

(Applying my code suggestion in GitHub directly will break a bunch of things. You should fix this locally.)